src/features/image-map-editor/image-canvas.tsx

```diff
@@ -0,0 +1,201 @@
+import { nanoid } from 'nanoid';
+import {
+  useRef,
+  useEffect,
+  useCallback,
+  MouseEvent,
+} from 'react';
+import { useEditor } from './editor.context';
+import { detectIsOverlap } from './utils';
+
+export type DrawStartCallback = () => void;
+export type DrawStopCallback = () => void;
+interface ImageCanvasProps {
+  width: number;
+  height: number;
+  onDrawStart?: DrawStartCallback;
+  onDrawStop?: DrawStopCallback;
+}
+
+const ImageCanvas = ({
+  width,
+  height,
+  onDrawStart,
+  onDrawStop,
+  ...rest
```

Comment:
I think this is an empty object. Any reason to pass it down?

Reply:
I set the data attribute on the element and styling on the parent scope.
Should I remove the `...rest` piece for a better understanding?

Reply:
Thanks for the suggestion, it's clear and helpful.